airflow/jobs/backfill_job.py

```diff
@@ -865,13 +863,7 @@ def reset_state_for_orphaned_tasks(self, filter_by_dag_run=None, session=None):
         if filter_by_dag_run is None:
             resettable_tis = (
                 session.query(TaskInstance)
-                .join(
-                    DagRun,
-                    and_(
-                        TaskInstance.dag_id == DagRun.dag_id,
-                        TaskInstance.execution_date == DagRun.execution_date,
-                    ),
-                )
+                .join(TaskInstance.dag_run)
```

Comment:
Worth using `contains_eager` here too?

Reply:
Yes probably -- didn't know about that when I first started.